autoload/lightline/bufferline.vim

```diff
@@ -71,14 +72,16 @@ function! s:get_buffer_name(i, buffer, path)
       let l:name = pathshorten(l:name)
     endif
   endif
-  if s:enable_devicons == 1 && exists('*WebDevIconsGetFileTypeSymbol')
+  if exists('*WebDevIconsGetFileTypeSymbol') && s:enable_devicons == 1 && s:reverse_buffer_icons == 0
     let l:name = WebDevIconsGetFileTypeSymbol(fnamemodify(bufname(a:buffer), ':t')) . ' ' . l:name
+  elseif s:reverse_buffer_icons == 1
+    let l:name = l:name . ' ' . WebDevIconsGetFileTypeSymbol(fnamemodify(bufname(a:buffer), ':t'))
   elseif s:enable_devicons == 1 && has('nvim-0.5') && exists('g:nvim_web_devicons')
     let l:name = v:lua._bufferline_get_icon(bufname(a:buffer)) . ' ' . l:name
   elseif s:enable_nerdfont == 1
     try
       let l:name = nerdfont#find(fnamemodify(bufname(a:buffer), ':t'), 0) . ' ' . l:name
-    catch /^Vim\%((\a\+)\)\=:E117:/
+      catch /^Vim\%((\a\+)\)\=:E117:/
     endtry
   endif
```

Comment:
Your lines 75-86 should be replaced with this code which retrieves an icon and applies the positioning afterwards:
```viml
  let l:icon = s:get_icon(a:buffer)
  if l:icon != ''
    let l:name = s:reverse_buffer_icons == 0 ? (l:icon . ' ' . l:name) : (l:name . ' ' . l:icon)
  endif
```
The icon fetching code should be refactored into a new function:
```viml
function! s:get_icon(buffer)
  if s:enable_devicons == 1 && exists('*WebDevIconsGetFileTypeSymbol')
    return WebDevIconsGetFileTypeSymbol(fnamemodify(bufname(a:buffer), ':t'))
  endif
  
  if s:enable_devicons == 1 && has('nvim-0.5') && exists('g:nvim_web_devicons')
    return v:lua._bufferline_get_icon(bufname(a:buffer))
  endif

 if s:enable_nerdfont == 1
    try
      return nerdfont#find(fnamemodify(bufname(a:buffer), ':t'), 0)
    catch /^Vim\%((\a\+)\)\=:E117:/
    endtry
  endif

  return ''
endfunction
```

This way the reverse buffer icons functionality works for all icon integrations.

Reply:
Thanks so much for pointing out my mistakes and for explaining in detail how to fix them.